base/markdown/GitHub/GitHub.jl

```diff
@@ -33,7 +33,83 @@ function github_paragraph(stream::IO, md::MD, config::Config)
     return true
 end
 
-# TODO: tables
+typealias Row Vector
 
-@flavor github [list, indentcode, blockquote, fencedcode, hashheader, github_paragraph,
-                 en_dash, inline_code, asterisk_bold, asterisk_italic, image, link]
+type Table
+    rows::Vector{Row}
+    align
+end
+
+function github_table(stream::IO, md::MD, config::Config)
+    withstream(stream) do
+        rows = Row[]
+        n = 0
+        align = :r # default is to align right
+        while !eof(stream)
+            n += 1
+            pos = position(stream)
+            skipwhitespace(stream)
+            line = readline(stream) |> chomp
+
+            if n == 1
+                pipe_border = line[1] == '|'
+                if !('|' in line)
+                    return false
+                end
+            end
+
+            row = map(strip, split(line, "|"))
+            if pipe_border
+                if row[1] == row[end] == ""
+                    row = row[2:end-1]
+                else
+                    return false
+                end
+            end
+
+            if n == 2 && all(['-' in r && issubset(Set(r), Set(" -:"))
+                             for r in row])
+                # handle possible --- line
+                align = Symbol[]
+                for r in row
+                    if r[1] == ':'
+                        if r[end] == ':'
+                            push!(align, :c)
+                        else
+                            push!(align, :l)
+                        end
+                    else
+                        if r[end] == ':'
+                            push!(align, :r)
+                        else
+                            # default is align right
+                            push!(align, :r)
+                        end
+                    end
+                end
+
+            elseif n == 1 || length(rows[1]) == length(row)
+                push!(rows, Row(map(x -> parseinline(IOBuffer(_full(x)), config), row)))
```

Comment:
You shouldn't need the `IOBuffer` constructor here, since `parseinline` has backup methods for strings. You might be able to get away with the `_full`, too, and if not I think it should be `parseinline`'s job to do the conversion.